### PR TITLE
misc(build): format build-tracker results and post to gh status api

### PR DESCRIPTION
**Pull request by paulirish**
output in action will look more like this
image

and status looks like

image

buildtracker went 1.0 a bit ago and so I've also upgraded it and pulled in a few more subpackages i need for formatting.

ref #10472

### CODE REVIEW

File: build-tracker.config.js [hunk 1]
```diff
@@ -14,4 +26,100 @@ module.exports = {
...
+
+  const summary = [hasWarnings ? '⚠️  See report.' : undefined, summaryStats]
+    .filter(s => s !== undefined)
+    .join(' ');
+
+  console.log(`\n${summary}`);
+
+  return postGHStatusForSha(
+    {
+      // Note: we don't use the `build` revision because—in a pull request build—this will
```
Comment (patrickhulce) [inline build-tracker.config.js:90]: this is a comment about not using the build revision, but it looks like we are? is `build.meta.revision` not the "`build` revision"?

File: build-tracker.config.js [hunk 2]
```diff
@@ -14,4 +26,100 @@ module.exports = {
...
+      // represent a merge commit "between the source branch and the upstream branch". This commit
+      // doesn't actually exist in the PR.
+      // https://docs.travis-ci.com/user/pull-requests/#how-pull-requests-are-built
```
Comment (patrickhulce) [inline build-tracker.config.js:93]: GitHub Actions link?

File: build-tracker.config.js [hunk 3]
```diff
@@ -14,4 +26,100 @@ module.exports = {
...
+    body: JSON.stringify({
+      state,
+      target_url: targetUrl,
+      description,
+      context: 'Build Tracker',
+    }),
+  }).then(async response => {
+    if (response.ok) return;
+    const json = await response.json();
+    throw new Error(`GH API not OK. Status: ${response.status}. Body: ${JSON.stringify(json)}`);
```
Comment (patrickhulce) [inline build-tracker.config.js:123]:
```suggestion
    throw new Error(`GH API not OK. Status: ${response.status}. Body: ${await response.text()}`);
```

**Review by patrickhulce — COMMENTED** [review]
it doesn't look like this posted a status check (see https://github.com/GoogleChrome/lighthouse/pull/12925/checks?check_run_id=3354584808#step:22:22) should we use the built-in build token to actions instead?

**Comment by paulirish** [comment]
> lets see if this still works

since the heroku app is down theres not much point in getting the gh status posting working IMO. (even tho we could..)

https://github.com/GoogleChrome/lighthouse/issues/10472#issuecomment-1865015002

**Pull request closed without merging**